content/_data/upgrades/2-221-1.adoc

```diff
@@ -0,0 +1,76 @@
+==== Header and Breadcrumb Layout Improvements
+
+Instances with plugins that depend on specific details of the Jenkins layout (like the https://plugins.jenkins.io/simple-theme-plugin/[Simple Theme plugin]) may have user interface issues with this release.
+In the case of the Simple Them Plugin, users or theme maintainers may need to update CSS and JavaScript to match the new layout structure.
+
+==== Removed Agent Protocols
+
+Remoting changelogs for https://github.com/jenkinsci/remoting/releases/tag/remoting-4.2[4.2],
+https://github.com/jenkinsci/remoting/releases/tag/remoting-4.0.1[4.0.1],
+https://github.com/jenkinsci/remoting/releases/tag/remoting-4.0[4.0],
+https://github.com/jenkinsci/remoting/releases/tag/remoting-3.40[3.40],
+https://github.com/jenkinsci/remoting/releases/tag/remoting-3.39[3.39].
+https://github.com/jenkinsci/remoting/releases/tag/remoting-3.38[3.38], and
+https://github.com/jenkinsci/remoting/releases/tag/remoting-3.37[3.37]
+
+The Remoting library has been updated to resolve a security issue in remoting versions prior to 4.0.1 and to add support for the <<websocket-protocol,WebSocket protocol>>.
+The update removes outdated agent protocols that had previously been deprecated.
+`Inbound TCP Agent Protocols` versions 1, 2, and 3 have been removed.
+
+<<websocket-protocol,WebSocket protocol>> connections require agents using Remoting 4.2 or later.
+
+plugin:ssh-slaves[SSH Build Agents] will always use the latest `agent.jar`, but other agent types may need to have their `agent.jar` updated manually.
+The following agent types should be updated:
+
+* Inbound TCP Agent Protocol agents (formerly known as Java Web Start agents) with `agent.jar` / `slave.jar` on the disk should update to version 4.2 or newer.
+  The latest `agent.jar` can be downloaded from any Jenkins server at `/jnlpJars/agent.jar` as in \https://your-jenkins-server.example.com/jnlpJars/agent.jar
+* plugin:swarm[Swarm Client] should update to version 3.18 or newer
+* Agent Docker packages offered by the Jenkins project (https://hub.docker.com/r/jenkins/slave/[jenkins/slave] and https://hub.docker.com/r/jenkins/jnlp-slave/[jenkins/jnlp-slave]) should update to version https://github.com/jenkinsci/docker-slave/releases/tag/4.0.1-1/[4.0.1-1] or newer
```

Comment:
Version is not correct. Any version with JNLP4 support will do the job

Reply:
@jeffret-b can you help me with alternate phrasing here?  

We specifically used "should" rather than "must" in this context so that we allowed for other versions without specifying the other versions in detail.

Reply:
We need to recommend that they upgrade to 4.0.1 or later so that they receive the fix for SECURITY-1659. They don't have to, but installing all security updates is recommended. That's why we chose "should". Not required, but recommended.

Reply:
SECURITY-1659 is a master-side Security issue, isn't it? It is not relevant to agent upgades unless I am missing something

Reply:
Hmm ... now I understand the confusion. Reading the [security advisory](https://jenkins.io/security/advisory/2020-01-29/) again I see that we weren't as clear as we should have been. @daniel-beck 

It's one of those cases of reading the thing multiple times and not seeing that it's leaving something out.

SECURITY-1660 is the master side. SECURITY-1659 is agent side, requiring code changes on the agent. They are very similar issues but in different places. The advisory content reads about the same for both and nothing in the advisory mentions updating agents or anything about Remoting.